Feature/collapsible example

ektasawant opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes https://jira-prod.tcc.etn.com/browse/PXBLUE-2384.

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- [X] Gradient effect on drawer header with menu icon
- [X] Desktop view with extended header and when scroll change it to collapsible view
- [X] Mobile view with extended header and when scroll change it to collapsible view
- [X] Show background image in extended header hide bg image in collapsible view

<!-- Include screenshots if they will help illustrate the changes in this PR -->
#### Screenshots:

Desktop:
![image](https://user-images.githubusercontent.com/10433274/129938040-bebff80a-a952-47ab-ba30-87397c363bd5.png)
Desktop collapsible:
![image](https://user-images.githubusercontent.com/10433274/129938584-c48d42cb-e4fd-4902-8489-5a65d034c46a.png)

Mobile:
![image](https://user-images.githubusercontent.com/10433274/129938705-32d20b04-ce42-49c1-a13b-c410e9270a5a.png)

Mobile collapsible
![image](https://user-images.githubusercontent.com/10433274/129938797-9f051902-77df-4305-afc2-5caa32699f64.png)

<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- http://localhost:3000/collapsible

ektasawant commented:
@huayunh , as per Figma comment https://www.figma.com/file/ZzZJSoGjeKdrYct57EMPS0?node-id=474:120228#102003981
I have added close icon in mobile view which closes the drawer. Please review

![image](https://user-images.githubusercontent.com/10433274/130052403-73ddc9d1-5f8e-48e6-80d8-759590e480ba.png)

jeffvg commented:
I'm getting this warning when I view in responsive view and drawer is hidden and collapsible appbar is collapsed and navigating to another item in the drawer.
![image](https://user-images.githubusercontent.com/37150565/130235804-99727801-e8cb-48e6-a4a8-2ef63ccef4f4.png)

daileytj commented:
> I'm getting this warning when I view in responsive view and drawer is hidden and collapsible appbar is collapsed and navigating to another item in the drawer.
> ![image](https://user-images.githubusercontent.com/37150565/130235804-99727801-e8cb-48e6-a4a8-2ef63ccef4f4.png)

I'm seeing the same thing in our React storybook. This is probably an issue with the underlying `AppBar` component. Let's log an issue for this in the `react-component-library` repo.

joebochill commented:
> I'm seeing the same thing in our React storybook. This is probably an issue with the underlying `AppBar` component. Let's log an issue for this in the `react-component-library` repo.

It's a storybook issue...there's already an issue report for it: https://github.com/pxblue/react-component-library/issues/154

jeffvg reviewed (approved):
looks good

huayunh commented:
> @huayunh , as per Figma comment https://www.figma.com/file/ZzZJSoGjeKdrYct57EMPS0?node-id=474:120228#102003981
I have added close icon in mobile view. Please review.

Would you mind moving the close button to the left of the text?

ektasawant commented:
> > @huayunh , as per Figma comment https://www.figma.com/file/ZzZJSoGjeKdrYct57EMPS0?node-id=474:120228#102003981
> > I have added close icon in mobile view. Please review.
> 
> Would you mind moving the close button to the left of the text? Like this: https://pxblue.github.io/patterns/navigation#temporary

Hi @huayunh , Please review below screenshot . Removed gradient effect to match https://pxblue.github.io/patterns/navigation#temporary. Let me know if we need to keep gradient effect.

![image](https://user-images.githubusercontent.com/10433274/130624898-4d89f895-816c-4f1f-9de5-118ed27de6d1.png)

huayunh commented:
Did you follow https://pxblue.github.io/patterns/appbar#mobile ? The vertical alignment of the three liner can be tricky

ektasawant commented:
> Did you follow https://pxblue.github.io/patterns/appbar#mobile ? The vertical alignment of the three liner can be tricky

![image](https://user-images.githubusercontent.com/10433274/130774774-d313d7a1-dcfa-4899-88ff-71c0a753f715.png)
For spacing between three liner I added `margin-top: -4px` which is working fine. Anything else that I need to check?

ektasawant commented:
> > Did you follow https://pxblue.github.io/patterns/appbar#mobile ? The vertical alignment of the three liner can be tricky
> 
> ![image](https://user-images.githubusercontent.com/10433274/130774774-d313d7a1-dcfa-4899-88ff-71c0a753f715.png)
> For spacing between three liner I added `margin-top: -4px` which is working fine. Anything else that I need to check?

![image](https://user-images.githubusercontent.com/10433274/130798351-368da4fc-43e2-4634-942b-5b8bcf2ed287.png)
Followed https://pxblue-components.github.io/angular/?path=/story/components-app-bar--with-full-config for increasing vertical space.

huayunh commented:
That looks better visually
